app.py

```diff
@@ -1,26 +1,402 @@
+                 search_hashtags: List,
+                 search_urls: List,
+                 remove_sentiment: str,
+                 certainty_low: float,
+                 certainty_high: float):
+    """
+    Main method that ties all the components together. Takes use of above helper methods.
+    """
+    # prepare model for analysis
     train_model_if_necessary()
+    # Verify data
+    start_date, end_date = dates
+    if (start_date > end_date):
+        raise ValueError(f'Start date {start_date} may not be later than end date {end_date}')
+    # negate bool as to get meaning worthy of get_tweets
+    # (app asks, "do you want local search?" get_tweets asks, "do you want online search?")
+    fresh_search = not fresh_search
+
+    # Scrape data
+    tweet_list = get_tweets(tweet_amount, fresh_search, hashtags)
+    print("Done scraping...")
+
+    # Preprocess data
+    clean_tweets, hashtag_stats, mention_stats = handle_tweet_data(tweet_list)
+    print("Done preprocessing...")
+
+    # analyze the clean data
+    analyzed_tweets = analyze_many_tweets(
+        clean_tweets, certainty_low, certainty_high)
+    print("Done analyzing...")
+
+    # filter data to specifics
+    filtered_data = _filter_data(
+        analyzed_tweets, start_date, end_date, search_hashtags, search_mentions, search_urls)
+    print('Done filtering data...')
+
+    # filter sentiment
+    if (remove_sentiment):
+        filtered_data = remove_sentiment(filtered_data, remove_sentiment)
+        print("Done removing sentiment...")
+
+    # Getting plot data from the get_sentiment function
+    plot_data = get_sentiment(filtered_data)
+    print("Done getting sentiment df for plotting...")
+
+    # Warn user that plt.show() is blocking
+    if not save_plot:
+        print('\tNOTICE\n\tShowing the plot will block the main thread.\n\tExit the plot display to continue program.')
+
+    # Create plot
+    file_name = '_'.join(hashtags)
+    if plot_type == "bar":
+        if save_plot:
+            bar_plot(plot_data, file_name, file_name)
+        else:
+            bar_plot(plot_data, file_name)
+    if plot_type == "line":
+        if save_plot:
+            line_plot(plot_data, file_name, file_name)
+        else:
+            line_plot(plot_data, file_name)
+    if plot_type == "pie":
+        if save_plot:
+            pie_chart(plot_data, file_name, file_name)
+        else:
+            pie_chart(plot_data, file_name)
+    # print statistics
+    print('--------------------------------------')
+    print('Printing statistics')
+    print('\tTOP 5 HASHTAGS')
+    pprint(list(hashtag_stats.items())[:5], width=1)
+    print('--------------------------------------')
+    print('\tTOP 5 MENTIONS')
+    pprint(list(mention_stats.items())[:5], width=1)
+    print('--------------------------------------')
+    print('\t EXAMPLE OBJECT')
+    pprint(analyzed_tweets[0], width=1)
+    print('--------------------------------------')
+
+
+if __name__ == "__main__":
```

Comment:
So this is basically "if using CLI" ?

Reply:
Not necessarily just for CLI. If you e.g. start a Flask Server in a terminal this if-statement is also needed

Reply:
Ok, yeah, so its "If run from terminal" then? :)

Reply:
Here is a very good walkthrough of the definition and its other values:
https://stackoverflow.com/a/419185

Essentially, __name__ is "__main__" when the program is the main program. If it is called by something else, as a module would, it is typically not "__main__".

Maybe if you allowed Presentation.py to be run in two setups, either as a standalone, or as a module. When run as a module, you call each function and use it. But you can and should specify its execution order in that statement (*if __name__ == "__main__":*)
![image](https://user-images.githubusercontent.com/37186286/82307242-5bbb6600-99c0-11ea-987a-524738971446.png)